### PR TITLE
Add unit tests for EncoderNLS / DecoderNLS fallbacks

### PR DESCRIPTION
**Pull request by GrabYourPitchforks**
Contributes to https://github.com/dotnet/corefx/issues/37969. Unit tests will not pass until https://github.com/dotnet/coreclr/pull/25397 is merged and makes its way over to corefx.

### CODE REVIEW

File: src/System.Text.Encoding/tests/Decoder/Decoder.cs [hunk 1]
```diff
@@ -223,6 +223,38 @@ public static void DecoderExceptionFallbackBufferTest()
             Assert.Equal('\u0000', fallbackBuffer.GetNextChar());
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".NET Framework contains known bugs in this code path.")]
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/Decoder/Decoder.cs]: @ViktorHofer, based on your previous comments, this should not be added?

Comment (ViktorHofer) [reply src/System.Text.Encoding/tests/Decoder/Decoder.cs]: If the test project doesn't have a netfx configuration then you don't need that line.

Comment (GrabYourPitchforks) [reply src/System.Text.Encoding/tests/Decoder/Decoder.cs]: The test project only has _netcoreapp_ and _uap_ configurations. I'll remove the line.

**Comment by GrabYourPitchforks** [comment]
Failing tests are unrelated. Going forward with merge.

**Pull request merged**